### PR TITLE
Fix fetch interface and fromInterface

### PR DESCRIPTION
**Pull request by samuel-williams-da**
Package maps for views now include packages used only for interface fetches, this fixed fetching an interface
fromInterface will now run upgrade conversions if the qualified name and package name match

### CODE REVIEW

**Comment by samuel-williams-da** [comment]
issue is that reinterpretation derives its package map purely from interface exercises, not interface fetches.
fetch nodes don't currently give an easy way to know if they're for interfaces
could consider an output in engine metadata to give this info, or even derive most of the package map entries in the engine

File: sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala [hunk 1]
```diff
@@ -1304,13 +1306,29 @@ private[lf] object SBuiltin {
...
+        (oTplIdPkgName, oTyConPkgName) match {
+          case (Some(tplIdPkgName), Some(tyConPkgName)) if tplIdPkgName == tyConPkgName =>
+            importValue(machine, tplId, record.toUnnormalizedValue) { templateArg =>
+              Control.Value(SOptional(Some(templateArg)))
+            }
+          case _ => Control.Value(SOptional(None))
+        }
       } else {
-        SOptional(None)
+        Control.Value(SOptional(None))
```
Comment (simonmaxen-da) [inline sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala:1331]: Not `Control.Error`?

**Review by simonmaxen-da — APPROVED** [review]
LGTM

File: sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala (hunk 1 above)
Comment (samuel-williams-da) [reply sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala:1331]: No, explicitly not that - fromInterface should give back None if the given template is not the "same" as what it holds, where "same" here means either same template id (+package id), or same qualified name + same package name + both support upgrades.
If there is an upgrade failure, the importValue will error.

**Comment by samuel-williams-da** [comment]
Going to revert the canton package map changes and fix to fetch interface, such that PR just includes the new Fetch node field, and the fix to `fromInterface`

**Pull request merged**